kinisi/diffusion.py

```diff
@@ -205,6 +206,53 @@ def compute_covariance_matrix(self) -> sc.Variable:
                             minimum_eigenvalue_method(cov[self.diff_regime:, self.diff_regime:], self._cond_max)),
                         unit=self.msd.unit**2)
 
+    def posterior_predictive(self,
+                             n_posterior_samples: int = None,
+                             n_predictive_samples: int = 256,
+                             progress: bool = True) -> sc.Variable:
+        """
+        Sample the posterior predictive distribution. The shape of the resulting array will be
+        `(n_posterior_samples * n_predictive_samples, start_dt)`.
+        
+        :param n_posterior_samples: Number of samples from the posterior distribution.
+            Optional, default is the number of posterior samples.
+        :param n_predictive_samples: Number of random samples per sample from the posterior distribution.
+            Optional, default is :py:attr:`256`.
+        :param progress: Show tqdm progress for sampling. Optional, default is :py:attr:`True`.
+
+        :return: Samples from the posterior predictive distribution. 
+        """
+        if n_posterior_samples is None:
+            n_posterior_samples = self.gradient.size
+
+        ppd_unit = self.gradient.unit * self.msd.coords['time interval'].unit + self.intercept.unit
+
+        diff_regime = np.argwhere(self.msd.coords['time interval'] >= self._start_dt)[0][0]
+        ppd = sc.zeros(
+            dims=['posterior samples', 'predictive samples', 'time interval'],
+            shape=[n_posterior_samples, n_predictive_samples, self.msd.coords['time interval'][diff_regime:].size],
+            unit=ppd_unit)
+        samples_to_draw = list(enumerate(np.random.randint(0, self.gradient.size, size=n_posterior_samples)))
+
+        if progress:
+            iterator = tqdm(samples_to_draw, desc='Calculating Posterior Predictive')
+        else:
+            iterator = samples_to_draw
+
+        # Testing unit consistency for mu and covariance before producing mv
+        try:
+            ppd_unit**2 == self._covariance_matrix.unit
```

Comment:
Is this ever not true?

Reply:
Wasn't sure, just thought better safe than sorry as we use the values directly in the multivariate normal